sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveTableScanExec.scala

```diff
@@ -62,6 +62,8 @@ case class HiveTableScanExec(
 
   override def conf: SQLConf = sparkSession.sessionState.conf
 
+  override def nodeName: String = s"${super.nodeName} (${relation.tableMeta.qualifiedName})"
```

Comment:
Our `DataSourceScanExec` is using `unquotedString` in nodeName. We need to make these LeafNode consistent. Could you check all the other `LeafExecNode`?

Reply:
[DataSourceScanExec](https://github.com/apache/spark/blob/ee56fc3432e7e328c29b88a255f7e2c2a4739754/sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala#L47) is using `unquotedString` and [so does this PR](https://github.com/apache/spark/blob/da35574297d96eb750bdfee755a48defc36e284a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala#L265). Rest implementations of `LeafExecNode` do not override `nodeName` or depend on base class (ie. `TreeNode`)

Reply:
How about 
> s"Scan HiveTable ${relation.tableMeta.qualifiedName}"

Reply:
`DataSourceV2ScanExec` faces the same issue. How about `InMemoryTableScanExec`?

Reply:
@gatorsmile : I have updated the PR after going through all the *ScanExec implementations

Changes introduced in this PR:

Scan impl | overridden `nodeName`
------------ | -------------
DataSourceV2ScanExec | `Scan DataSourceV2 [output_attribute1, output_attribute2, ..]`
ExternalRDDScanExec | `Scan ExternalRDD [output_attribute1, output_attribute2, ..]`
FileSourceScanExec | `Scan FileSource ${tableIdentifier.map(_.unquotedString).getOrElse(relation.location)}"`
HiveTableScanExec | `Scan HiveTable relation.tableMeta.qualifiedName`
InMemoryTableScanExec | `Scan In-memory relation.tableName`
LocalTableScanExec | `Scan LocalTable [output_attribute1, output_attribute2, ..]`
RDDScanExec | `Scan RDD name [output_attribute1, output_attribute2, ..]`
RowDataSourceScanExec | `Scan FileSource ${tableIdentifier.map(_.unquotedString).getOrElse(relation)}`


Things not affected:
- DataSourceScanExec : already uses `Scan relation tableIdentifier.unquotedString`
- RDDScanExec forces clients to specify the `nodeName`